contracts/car-races/io/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub struct GameInit {
+    pub config: Config,
+}
+
+#[derive(Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub enum GameAction {
+    AddAdmin(ActorId),
+    RemoveAdmin(ActorId),
+    AddStrategyIds {
+        car_ids: Vec<ActorId>,
+    },
+    StartGame,
+    Play {
+        account: ActorId,
+    },
+    PlayerMove {
+        strategy_action: StrategyAction,
+    },
+    UpdateConfig {
+        add_attribute_gas: Option<u64>,
+        tokens_for_owner_gas: Option<u64>,
+        gas_to_remove_game: Option<u64>,
+        initial_speed: Option<u32>,
+        min_speed: Option<u32>,
+        max_speed: Option<u32>,
+        gas_for_round: Option<u64>,
+        time_interval: Option<u32>,
+        max_distance: Option<u32>,
+        time: Option<u32>,
+        time_for_game_storage: Option<u64>,
+    },
+    RemoveGameInstance {
+        account_id: ActorId,
+    },
+    RemoveGameInstances {
+        players_ids: Option<Vec<ActorId>>,
+    },
+    AllowMessages(bool),
+}
+
+#[derive(Encode, Decode, TypeInfo, Debug)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub enum StrategyAction {
+    BuyAcceleration,
+    BuyShell,
+    Skip,
+}
+
+#[derive(Encode, Decode, TypeInfo, Debug, Clone, PartialEq, Eq)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub enum RoundAction {
+    Accelerated,
+    SlowedDown,
+    SlowedDownAndAccelerated,
+}
+
+#[derive(Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub enum CarAction {
+    YourTurn(BTreeMap<ActorId, Car>),
+}
+
+#[derive(Encode, Decode, TypeInfo, PartialEq, Eq)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub enum GameReply {
+    GameStarted,
+    NotEnoughGas,
+    GameFinished,
+    GasReserved,
+    StrategyAdded,
+    PlayersMove,
+}
+
+impl Game {
+    pub fn buy_acceleration(&mut self) {
+        let car_id = self.get_current_car_id();
+        let car = self.cars.get_mut(&car_id).expect("Get Car: Can't be None");
+        car.speed += DEFAULT_SPEED;
+        car.car_actions.push(RoundAction::Accelerated);
+    }
+
+    pub fn buy_shell(&mut self) {
+        let car_id = self.get_current_car_id();
+
+        let shelled_car_id = self.find_car_to_shell(&car_id);
+        self.cars.entry(shelled_car_id).and_modify(|car| {
+            let new_speed = car.speed - DEFAULT_SPEED;
+            if new_speed < DEFAULT_SPEED {
+                car.speed = DEFAULT_SPEED;
+            } else {
+                car.speed = new_speed;
+            }
+            car.car_actions.push(RoundAction::SlowedDown);
+        });
+    }
+
+    pub fn get_current_car_id(&self) -> ActorId {
+        self.car_ids[self.current_turn as usize]
+    }
+
+    pub fn update_positions(&mut self, config: &Config) {
+        let mut winners = Vec::new();
```

Comment:
Could we use capacity equal to 3 here?
```suggestion
        let mut winners = Vec::with_capacity(3);
```